lib/assets/javascripts/cartodb/models/filter.js

```diff
@@ -579,6 +579,10 @@ cdb.admin.models.Filters = Backbone.Collection.extend({
       throw "Filters need a table";
     }
     this.table = options.table;
+    if (!options.dataLayer) {
+      throw "Filters need a dataLayer";
+    }
+    this.dataLayer = options.dataLayer;
```

Comment:
you should use, add_related_model in order to avoid to have leaks https://github.com/CartoDB/cartodb.js/blob/2983b2fdcef914afdb1f4fdae173471143930452/src/core/view.js#L19